subgraphs/compound-forks/protocols/aurigami/src/mapping.ts

```diff
@@ -302,11 +323,138 @@ function getOrCreateProtocol(): LendingProtocol {
     "Aurigami",
     "aurigami",
     "2.0.1",
-    "1.1.6",
+    "1.1.7",
     "1.0.0",
     Network.AURORA,
     comptroller.try_liquidationIncentiveMantissa(),
     comptroller.try_oracle()
   );
   return _getOrCreateProtocol(protocolData);
 }
+
+/////////////////
+//// Helpers ////
+/////////////////
+
+function getPrice(
+  marketAddress: Address,
+  priceOracle: string
+): ethereum.CallResult<BigInt> {
+  //
+  //
+  // There are 4 markets where the price oracle does not work
+  // PLY, AURORA, TRI, USN
+  // Use Trisolaris LP pool pairs to derive price
+
+  if (marketAddress == PLY_MARKET) {
+    return getPriceFromLp(
+      priceOracle,
+      WNEAR_MARKET,
+      PLY_MARKET,
+      WNEAR_PLY_LP,
+      true
+    );
+  }
+  if (marketAddress == AURORA_MARKET) {
+    return getPriceFromLp(
+      priceOracle,
+      ETH_MARKET,
+      AURORA_MARKET,
+      AURORA_ETH_LP,
+      true
+    );
+  }
+  if (marketAddress == TRI_MARKET) {
+    return getPriceFromLp(
+      priceOracle,
+      USDT_MARKET,
+      TRI_MARKET,
+      TRI_USDT_LP,
+      false
+    );
+  }
+  if (marketAddress == USN_MARKET) {
+    return getPriceFromLp(
+      priceOracle,
+      WNEAR_MARKET,
+      USN_MARKET,
+      WNEAR_USN_LP,
+      true
+    );
+  }
+
+  // get the price normally
+  const oracleContract = PriceOracle.bind(Address.fromString(priceOracle));
+  return oracleContract.try_getUnderlyingPrice(marketAddress);
+}
+
+//
+//
+// Gets the price of wantAddress.inputToken from an LP from Trisolaris
+function getPriceFromLp(
+  priceOracle: string, // aurigami price oracle
+  knownMarketID: Address, // address of the market we know the price of
+  wantAddress: Address, // market address of token we want to price
+  lpAddress: Address, // address of LP token
+  whichPrice: boolean // true to find token0 price in LP, false for token1
```

Comment:
I don't 100% understand why you need to tell which token from the pool to get the price from. Isn't that redundant if it will always be `wantAddress.inputToken`?

If wantAddress is token0, then you know you want the price for token0. Otherwise it's token1. Having that boolean could make someone accidentally pass it a wrong value resulting in inverted prices

Reply:
good point, changing this